### PR TITLE
[FIX] Fixes wrong visual range for ToggleTodo

### LINKED ISSUE #178: [BUG] ToggleTodo from normal mode sometimes takes last visual mode marks
**Please confirm**
- [x] I am running the latest version of this plugin

**Describe the bug**
When calling the function toggle_todo() from normal mode, the ToggleTodo function sometimes acts on the last visual selection, not the current line. 

**To Reproduce**
1. Start nvim
2. Open a telekasten note
3. Visually select some content.
4. Hit esc to cancel visual selection. (More generally, do something to return to normal mode, eg y to yank.)
5. Move to a different line.
6. Call `:lua require('telekasten').toggle_todo()<cr>`
7. toggle_todo() acts on the lines of the previous visual selection, not the current line.

**Expected behavior**
From normal mode, toggle_todo() should always act on the current line. 

**Operating system (please complete the following information):**
- MacOS 11.7
- NVIM 0.8.0

### PR DESCRIPTION
**Pull request by alex-laycalvert**
## Proposed change

This PR fixes the issue where selecting a visual range then doing `toggle_todo` on a different line will target the previously selected range. This works by passing a `v = true` option to the `toggle_todo` function that will let the function know to look for marks. This will be applied the visual mapping for the `toggle_todo`.

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update

## Additional information

- This PR fixes or closes issue: fixes #178 

## Checklist

- [x] I am running the **latest** version of the plugin.
- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Stylua (a `.stylua.toml` file is provided)
- [x] The code has been checked with luacheck (a `.luacheckrc` file is provided)
- [x] The `README.md` has been updated according to this change.
- [x] The `doc/telekasten.txt` helpfile has been updated according to this change.

### CODE REVIEW

File: doc/telekasten.txt [hunk 1]
```diff
@@ -670,13 +670,15 @@ telekasten.toggle_todo({opts})~
   to avoid neovim sending the wrong visual selection.
 
   Valid keys for {opts}
-
+ bv  z
```
Comment (lambtho12) [inline doc/telekasten.txt:673]: what is that?

**Review by lambtho12 — CHANGES_REQUESTED** [review]
It may be nice to add a line with the mapping for visual mode as well in the README and doc/telekasten.txt

```vimtext
nnoremap <leader>zt :lua require('telekasten').toggle_todo()<CR> 
vnoremap <leader>zt :lua require('telekasten').toggle_todo({v=true})<CR>
```

**Comment by lambtho12** [comment]
I agree with the solution proposed. I would have preferred a more seamless implementation, but I can't think of any either. So this will do.

Thank you very much for the fix and your reactivity. It is a delight to see other people getting involved in this!

File: doc/telekasten.txt (hunk 1 above)
Comment (lambtho12) [reply doc/telekasten.txt:673]: @alex-laycalvert what is this `bv z`? Is it a typo?

**Comment by alex-laycalvert** [comment]
No problem, just wish I had time to work on this earlier. I think the main issue with this is that after a visual range is selected and _something_ happens to it, the marks stay the same until a different range is selected. So we're kind of battling whether we know for certain if the visual marks set were legit or not.

File: doc/telekasten.txt (hunk 1 above)
Comment (alex-laycalvert) [reply doc/telekasten.txt]: Yes, fixed now

**Pull request merged**